apis/youtube.py

```diff
@@ -36,8 +36,9 @@ def search(self, search_query, max_length, min_view_count):
         return youtube_video_link
     
     def download(self, url, audio_bitrate):
-
         youtube_video = pytubeYouTube(url)
+        if youtube_video.age_restricted:
+            youtube_video.bypass_age_gate()
```

Comment:
This doesn't seem to work. Providing the oauth flow to pytubeYouTube also only worked once.

Need to look into it further to see if there's a long lived auth option that can be setup before the script is run.

Reply:
@CarvellWakeman Still looking through your PR, but for the most part looks good. Just to confirm, is it still possible in addition to still download songs without being logged in?

Reply:
Ah good point, no. Tekore can supports application tokens ([see guide](https://tekore.readthedocs.io/en/stable/auth_guide.html)) but I have not implemented it.

Should be trivial to support, just call `tk.request_client_token(clientId, clientSecret)` and pass the same Token object as user auth into `tk.Spotify(token)`

The hard(er) part is supporting both login types through some interface. The flask server in login.py only makes the user auth more convenient to avoid copying the redirectUri into the command line.

I'll think through it a bit.